tests/functional/UseCase/Workspace/CreateDropWorkspaceTest.php

```diff
@@ -0,0 +1,278 @@
+
+        /** @var array<string, string> $datasets */
+        $datasets = $bqClient->runQuery($bqClient->query(sprintf('SELECT
+  schema_name
+FROM
+  %s.INFORMATION_SCHEMA.SCHEMATA;', BigqueryQuote::quoteSingleIdentifier($projectId))))->getIterator()->current();
+
+        $this->assertSame(
+            strtoupper($response->getWorkspaceObjectName()),
+            $datasets['schema_name']
+        );
+
+        // test ws service acc is owner of ws dataset
+        $workspaceDataset = $bqClient->dataset($response->getWorkspaceObjectName())->info();
+        $this->assertNotNull($workspaceDataset);
+        $this->assertCount(1, $workspaceDataset['access']);
+        $this->assertSame('OWNER', $workspaceDataset['access'][0]['role']);
+        $this->assertSame($wsServiceAccEmail, $workspaceDataset['access'][0]['userByEmail']);
+
+        $cloudResourceManager = $this->clientManager->getCloudResourceManager($this->projectCredentials);
+        $actualPolicy = $cloudResourceManager->projects->getIamPolicy(
+            'projects/' . $projectId,
+            (new Google_Service_CloudResourceManager_GetIamPolicyRequest()),
+            []
+        );
+        $actualPolicy = $actualPolicy->getBindings();
+
+        $serviceAccRoles = [];
+        foreach ($actualPolicy as $policy) {
+            if (in_array('serviceAccount:' . $wsServiceAccEmail, $policy->getMembers())) {
+                $serviceAccRoles[] = $policy->getRole();
+            }
+        }
+
+        // ws service acc must have a job user role to be able to run queries
+        $expected = [
+            IAmPermissions::ROLES_BIGQUERY_DATA_VIEWER, // readOnly access
+            IAmPermissions::ROLES_BIGQUERY_JOB_USER,
+        ];
+        $this->assertEqualsArrays($expected, $serviceAccRoles);
+
+        try {
+            $wsBqClient->runQuery($wsBqClient->query(sprintf(
+                'CREATE SCHEMA `should_fail`',
+            )));
+            $this->fail('The workspace user should not have the right to create a new dataset.');
+        } catch (ServiceException $exception) {
+            $this->assertSame(403, $exception->getCode());
+            $this->assertStringContainsString(
+                'User does not have bigquery.datasets.create permission in project',
+                $exception->getMessage()
+            );
+        }
+
+        $bqClient->runQuery($bqClient->query(sprintf(
+            'CREATE SCHEMA `testReadOnlySchema`;',
+        )));
+
+        $bqClient->runQuery($bqClient->query(sprintf(
+            'CREATE TABLE `testReadOnlySchema`.`testTable` (`id` INTEGER);',
+        )));
+
+        $bqClient->runQuery($bqClient->query(sprintf(
+            'INSERT `testReadOnlySchema`.`testTable` (`id`) VALUES (1), (2), (3);',
+        )));
+
+        $result = $wsBqClient->runQuery($wsBqClient->query('SELECT * FROM `testReadOnlySchema`.`testTable`;'));
+
+        $this->assertCount(3, $result);
```

Comment:
bylo by super kdyby to přímo založilo bucket a tabulku přes cmd, ať to funguje v tom realném flow. Taky by to mohlo testovat jestli tam nemožeš zapsat, nebo dropnůt tu tabulku/dataset

Reply:
upravil som